src/test/java/org/apache/commons/collections4/bloomfilter/DefaultBloomFilterTest.java

```diff
@@ -67,6 +68,110 @@ public void testHasherBasedMergeWithDifferingSparseness() {
                 .forEachBitMapPair(bf1, (x, y) -> x == y));
     }
 
+    @Test
+    public void testEstimateNWithBrokenCardinality() {
+        // build a filter
+        BloomFilter filter1 = TestingHashers.populateEntireFilter(new BrokenCardinality(getTestShape()));
+        assertThrows(IllegalArgumentException.class, ()->filter1.estimateN());
+    }
+
+    @Test
+    public void testLargeBitmapAsArray() {
+        Shape s = Shape.fromKM(1, Integer.MAX_VALUE);
+        // create a very large filter with Integer.MAX_VALUE-1 bit set.
+        BloomFilter bf1 = new SimpleBloomFilter(s);
+        bf1.merge(new BitMapProducer() {
+            @Override
+            public boolean forEachBitMap(LongPredicate predicate) {
+                int limit = Integer.MAX_VALUE-1;
+                while (limit>64) {
+                    predicate.test(0xFFFFFFFFFFFFFFFFL);
+                    limit -= 64;
+                }
+                long last = 0L;
+                for (int i=0; i<limit; i++) {
+                    last |= BitMap.getLongBit(i);
+                }
+                predicate.test(last);
+                return true;
+            }} );
+        // the actual result of the calculation is: 46144189292
+        long[] ary = bf1.asBitMapArray();
+        for (int i=0; i<ary.length-2; i++) {
+            assertEquals(0xFFFFFFFFFFFFFFFFL, ary[i]);
+        }
+        int lastBits = (Integer.MAX_VALUE-1) % 64;
+        long last = 0L;
+        for (int i=0; i<lastBits; i++) {
+            last |= BitMap.getLongBit(i);
+        }
+        assertEquals(last, ary[ary.length-1]);
+    }
+
+    @Test
+    public void testEstimateLargeN() {
+        Shape s = Shape.fromKM(1, Integer.MAX_VALUE);
+        // create a very large filter with Integer.MAX_VALUE-1 bit set.
+        BloomFilter bf1 = new SimpleBloomFilter(s);
+        bf1.merge(new BitMapProducer() {
+            @Override
+            public boolean forEachBitMap(LongPredicate predicate) {
+                int limit = Integer.MAX_VALUE-1;
+                while (limit>64) {
+                    predicate.test(0xFFFFFFFFFFFFFFFFL);
+                    limit -= 64;
+                }
+                long last = 0L;
+                for (int i=0; i<limit; i++) {
+                    last |= BitMap.getLongBit(i);
+                }
+                predicate.test(last);
+                return true;
+            }} );
+        // the actual result of the calculation is: 46144189292
+        assertEquals(Integer.MAX_VALUE, bf1.estimateN());
+    }
+
+    @Test
+    public void testIntersectionLimit() {
+        Shape s = Shape.fromKM(1, Integer.MAX_VALUE);
+        // create a very large filter with Integer.MAX_VALUE-1 bit set.
+        BloomFilter bf1 = new SimpleBloomFilter(s);
+        bf1.merge(new BitMapProducer() {
+            @Override
+            public boolean forEachBitMap(LongPredicate predicate) {
+                int limit = Integer.MAX_VALUE-1;
```

Comment:
`limit` should be Integer.MAX_VALUE. There are a lot of whitespace issues here. See the comments for the method `testLargeBitmapAsArray`.